readthedocs/search/indexes.py

```diff
@@ -220,25 +218,24 @@ def get_mapping(self):
                 # Disable _all field to reduce index size.
                 '_all': {'enabled': False},
                 'properties': {
-                    'id': {'type': 'long'},
-                    'name': {'type': 'string', 'analyzer': 'default_icu'},
-                    'description': {'type': 'string', 'analyzer': 'default_icu'},
-
-                    'slug': {'type': 'string', 'index': 'not_analyzed'},
-                    'lang': {'type': 'string', 'index': 'not_analyzed'},
-                    'tags': {'type': 'string', 'index': 'not_analyzed'},
-                    'privacy': {'type': 'string', 'index': 'not_analyzed'},
+                    'id': {'type': 'keyword'},
```

Comment:
Is not `id` is `integer`?

Reply:
I believe the new ES doesn't let these vary types, or something. I was getting an error, and it doesn't really _need_ to be an ID, so we will just use it as a string like the others.

Reply:
@ericholscher Seems like they support it
https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html

Reply:
It does seem strange that we're coercing to string here, but I'm not familiar enough with ES to say why token/string is better.

Reply:
It's not that it's better, it just doesn't work otherwise. I don't fully understand it either.

Reply:
It throws this error: `RequestError: TransportError(400, u'illegal_argument_exception', u'mapper [id] cannot be changed from type [long] to [keyword]')`